packages/babel-helpers/src/helpers.js

```diff
@@ -523,8 +525,12 @@ helpers.isNativeReflectConstruct = helper("7.9.0")`
     try {
       // If the internal slots aren't set, this throws an error similar to
       //   TypeError: this is not a Date object.
-      Date.prototype.toString.call(Reflect.construct(Date, [], function() {}));
-      return true;
+      // Date#toString is generic in ES2015 [1] and it doesn't throw, but it
+      // returns "Invalid Date" if the receiver is not a properly initialized
+      // Date object.
+      // [1]: https://github.com/tc39/ecma262/issues/1268#issuecomment-410104832
+
+      return Date.prototype.toString.call(Reflect.construct(Date, [], function() {})) !== "Invalid Date";
```

Comment:
We could also change this to `Boolean.prototype.valueOf`:

- ES5: https://262.ecma-international.org/5.1/#sec-15.6.4.3
- ES6: https://262.ecma-international.org/6.0/#sec-boolean.prototype.valueof
- Next: https://tc39.es/ecma262/#sec-boolean.prototype.valueof